src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,59 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_SIZE];
+        values = (V[]) new Object[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = findKey(key);
+        if (index != -1 || isFull()) {
+            values[index] = value;
+            return;
```

Comment:
in this method you need not one "if" block , because you have to cases with different
 action :
 1) You have key, and change value ;
 2) You have a new pair(key, value) and need to add in your storage ;
 Your code should be something like that : 
 
```suggestion
        int index = findKey(key);
        if (index != -1) {
            values[index] = value;
            return;
         }
         if (!isFull()) {
            keys[size] = key;
            values[size] = value;
            size++;
        }
```